### PR TITLE
Fixed high CPU usage due to Socket.Select not blocking the thread properly when the game started

### PR DESCRIPTION
**Pull request by asl97**
As discussed in #7237.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
Same here as Paul.

File: OpenRA.Game/Server/Server.cs [hunk 1]
```diff
@@ -156,11 +156,11 @@ public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData)
 				for (;;)
 				{
 					var checkRead = new List<Socket>();
-					checkRead.Add(listener.Server);
+					if (State == ServerState.WaitingPlayers) checkRead.Add(listener.Server);
 					foreach (var c in Conns) checkRead.Add(c.socket);
 					foreach (var c in PreConns) checkRead.Add(c.socket);
 
-					Socket.Select(checkRead, null, null, timeout);
+					if (checkRead.Count > 0) Socket.Select(checkRead, null, null, timeout);
```
Comment (Phrohdoh) [inline OpenRA.Game/Server/Server.cs]: For both lines:
``` CSharp
if (foo)
    bar;
```

You can also use `checkRead.Any()`.

Comment (Mailaender) [inline OpenRA.Game/Server/Server.cs]: This whole file needs a StyleCop cleanup. Waiting for the students at scm.thm.de/openra/openra-develop to merge in theirs.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/5ade2bc6c9d12a0af7556ae8547adbef4ef6b9cf...4492d6b88e8612aaa4d31c9fd92de10c4d8cebe1)

**Comment by chrisforbes** [comment]
Nak, this is silly. What's actually going wrong in Mono's Select wrapper here?

**Comment by pchote** [comment]
The problem might be in glibc's select implementation.  The BUGS section in the man page explains that there are cases where a socket can be spuriously reported as ready.

OSX uses a BSD-based libc, which would explain why we didn't see this there.

**Comment by chrisforbes** [comment]
Those bugs won't be in play here -- definitely not constantly, which is
required for this to eat an entire core.

**Comment by pchote** [comment]
Good point.  In any case, it was rather bogus to go out of our way to pass a closed socket to Select, which is what this was fixing.  The problem would have been introduced by 3f0fafb3801a92d9ee6d5f3e73dcdeff6db31954.

**Comment by chrisforbes** [comment]
OK, I'd also misread the patch and seen something that looked like voodoo
chicken retrying. Patch is fine, closed listen socket explains this.

**Pull request merged**